src/runtime-rs/crates/runtimes/src/manager.rs

```diff
@@ -299,13 +299,22 @@ impl RuntimeHandlerManager {
         Ok(inner.get_kata_tracer())
     }
 
+    //init the sandbox for the normal task api
     #[instrument]
     async fn task_init_runtime_instance(
         &self,
         spec: &oci::Spec,
         state: &spec::State,
         options: &Option<Vec<u8>>,
     ) -> Result<()> {
+        let mut inner: tokio::sync::RwLockWriteGuard<'_, RuntimeHandlerManagerInner> =
+            self.inner.write().await;
+
+        // return if runtime instance has init
+        if inner.runtime_instance.is_some() {
+            return Ok(());
+        }
```

Comment:
`RuntimeHandlerManagerInner::try_init()` will run the same check when it's called later on at the bottom of this function - do we need to run it here as well?

To be clear, I'm not saying this is bad in some way, just making sure I understand.

Reply:
Hi @pmores 

Good question. There's two init function: one is task init and the other one is sandbox init. Task init is used for the legacy  api, and the sandbox init is used for the new sandbox api. Both of them need to check the runtime instance.
Since the RuntimeHandlerManagerInner::try_init() will also be called by sandbox init, thus it also need to check it. 

I moved the check from RuntimeHandlerManagerInner::try_init() into sandbox init to make the check looked  much more reasonable.